scheduler/test/cook/test/mesos/mesos_mock.clj

```diff
@@ -0,0 +1,844 @@
+(ns cook.test.mesos.mesos-mock
+  (:use clojure.test)
+  (:require [clojure.core.async :as async]
+            [clojure.data.csv :as csv]
+            [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [cook.mesos :as c]
+            [cook.mesos.mesos-mock :as mm]
+            [cook.mesos.share :as share]
+            [cook.test.testutil :refer (restore-fresh-database! create-dummy-group create-dummy-job create-dummy-instance)]
+            [datomic.api :as d]
+            [mesomatic.scheduler :as mesos]
+            [mesomatic.types :as mesos-types])
+  (:import org.apache.curator.framework.CuratorFrameworkFactory
+           org.apache.curator.framework.state.ConnectionStateListener
+           org.apache.curator.retry.BoundedExponentialBackoffRetry))
+
+(defn string->uuid
+  "Parses the uuid if `uuid-str` is a uuid, returns nil otherwise"
+  [uuid-str]
+  (try
+    (java.util.UUID/fromString uuid-str)
+    (catch Exception e
+      nil)))
+
+(defn noop-mock-scheduler
+  []
+  (mesos/scheduler
+    (registered [this driver framework-id master-info])
+    (reregistered [this driver master-info])
+    (offer-rescinded [this driver offer-id])
+    (framework-message [this driver executor-id slave-id data])
+    (disconnected [this driver])
+    (slave-lost [this driver slave-id])
+    (executor-lost [this driver executor-id slave-id status])
+    (error [this driver message])
+    (resource-offers [this driver offers])
+    (status-update [this driver status])))
+
+(deftest make-offer
+  (let [basic-host-info {:hostname "hosta"
```

Comment:
I will change where it makes sense. For this function, the offer is built based on the values of the host so we would need to specify everything about the host anyway.